scripts/windows/test/Run-E2ETest.ps1

```diff
@@ -0,0 +1,471 @@
+<#
+    .SYNOPSIS
+        Powershell utility to streamline running E2E tests for Windows.
+
+    .DESCRIPTION
+        It is used to wrap all related steps to run E2E tests for Windows;
+        it runs clean up, E2E test of given name, and print logs.
+    
+        To get details about parameters, please run "Get-Help .\Run-E2ETest.ps1 -Parameter *"
+        To find out what E2E tests are supported, just run "Get-Help .\Run-E2ETest.ps1 -Parameter TestName"
+        
+        Please ensure that E2E test folder have below folders/files:
+        - artifacts\core-windows: artifact from Image build.
+        - artifacts\iotedged-windows: artifact from edgelet build.
+        - artifacts\packages: contains packages of Moby docker engine, CLI and IoT Edge security daemon.
+          Either artifacts\iotedged-windows or packages folder exists, which is used for IoT Edge security daemon installation.
+        - certs: contains CA certs for certificate-related testing.
+
+    .PARAMETER E2ETestFolder
+        Path of E2E test folder
+
+    .PARAMETER ReleaseLabel
+        Release label, can be uniquely identify the build (e.g <ReleaseName>-<ReleaseAttempt>); which is used as part of Edge device name.
+
+    .PARAMETER ReleaseArtifactImageBuildNumber
+        Release artifact image build number; it is used to construct path of docker images, pulling from EdgeBuilds docker registry. E.g. 20190101.1.
+
+    .PARAMETER Architecture
+        Runtime architecture
+        Note: Valid values are "x64" and "arm32v7".
+
+    .PARAMETER TestName
+        Name of E2E test to be run
+        Note: Valid values are: "TempSensor", "TempFilter", "QuickstartCerts", "TransparentGateway".
+
+    .PARAMETER ContainerRegistryUserName
+        Username of container registry
+
+    .PARAMETER ContainerRegistryPassword
+        Password of given username for container registory
+
+    .PARAMETER IoTHubConnectionString
+        IoT hub connection string for creating edge device
+
+    .PARAMETER EventHubConnectionString
+        Event hub connection string for receive D2C messages
```

Comment:
Suggestion is to add the default value, of no parameter is passed.

Reply:
Replied in other comment.